proxy/grpcproxy/metrics.go

```diff
@@ -56,3 +66,43 @@ func init() {
 	prometheus.MustRegister(cacheHits)
 	prometheus.MustRegister(cachedMisses)
 }
+
+// HandleMetrics handler adds passthrough to endpoint '/metrics'.
+func HandleMetrics(mux *http.ServeMux, c *http.Client, eps []string) {
+	// random shuffle endpoints
+	r := rand.New(rand.NewSource(int64(time.Now().Nanosecond())))
+	eps = shuffleEndpoints(r, eps)
+
+	pathMetrics := etcdhttp.PathMetrics
+	target := fmt.Sprintf("%s%s", eps[0], pathMetrics)
+	if !strings.HasPrefix(target, "http") {
+		target = fmt.Sprintf("%s%s", "http://", target)
```

Comment:
We don't handle `https` here?

Reply:
reads if not `if !strings.HasPrefix(target, "http")`

if scheme = `https` target does not change as `http` matches `https` as well. If no scheme we assume http.

Reply:
But we are setting the scheme `http://`?

Reply:
https://play.golang.org/p/PENCHQL99GQ

Reply:
What if `"member:2379"` were for TLS endpoints? I remember scheme was not required for etcd gRPC endpoints.

Could you check?

Reply:
I think Get fails without scheme.

Reply:
> Could you check?

sounds good

Reply:
Thanks!

Reply:
Updated is that what you were thinking?